app/app.go

```diff
@@ -456,209 +475,101 @@ func New(
-		// check that we are at the height before the upgrade
-		if req.Height == app.upgradeHeightV2-1 {
-			app.BaseApp.Logger().Info(fmt.Sprintf("upgrading from app version %v to 2", currentVersion))
-			app.SetInitialAppVersionInConsensusParams(ctx, v2)
-			app.SetAppVersion(ctx, v2)
-
-			// The blobstream module was disabled in v2 so the following line
-			// removes the params subspace for blobstream.
-			if err := app.ParamsKeeper.DeleteSubspace(blobstreamtypes.ModuleName); err != nil {
-				panic(err)
-			}
-		}
-		// from v2 to v3 and onwards we use a signaling mechanism
-	} else if shouldUpgrade, newVersion := app.SignalKeeper.ShouldUpgrade(ctx); shouldUpgrade {
+func (app *App) EndBlocker(ctx sdk.Context) (sdk.EndBlock, error) {
+	res, err := app.ModuleManager.EndBlock(ctx)
+	if err != nil {
+		return sdk.EndBlock{}, err
+	}
+	currentVersion, err := app.AppVersion(ctx)
+	if err != nil {
+		return sdk.EndBlock{}, err
+	}
+
+	// use a signaling mechanism for upgrade
+	if shouldUpgrade, newVersion := app.SignalKeeper.ShouldUpgrade(ctx); shouldUpgrade {
 		// Version changes must be increasing. Downgrades are not permitted
 		if newVersion > currentVersion {
 			app.BaseApp.Logger().Info("upgrading app version", "current version", currentVersion, "new version", newVersion)
-			app.SetAppVersion(ctx, newVersion)
+			if err := app.SetAppVersion(ctx, newVersion); err != nil {
+				return sdk.EndBlock{}, err
+			}
 			app.SignalKeeper.ResetTally(ctx)
 		}
 	}
-	res.Timeouts.TimeoutCommit = app.getTimeoutCommit(currentVersion)
-	res.Timeouts.TimeoutPropose = appconsts.GetTimeoutPropose(currentVersion)
-	return res
-}
-
-// migrateCommitStore tells the baseapp during a version upgrade, which stores to add and which
-// stores to remove
-func (app *App) migrateCommitStore(fromVersion, toVersion uint64) (baseapp.StoreMigrations, error) {
-	oldStoreKeys := app.keyVersions[fromVersion]
-	newStoreKeys := app.keyVersions[toVersion]
-	result := baseapp.StoreMigrations{
-		Added:   make(map[string]*storetypes.KVStoreKey),
-		Deleted: make(map[string]*storetypes.KVStoreKey),
-	}
-	for _, oldKey := range oldStoreKeys {
-		if !slices.Contains(newStoreKeys, oldKey) {
-			result.Deleted[oldKey] = app.keys[oldKey]
-		}
-	}
-	for _, newKey := range newStoreKeys {
-		if !slices.Contains(oldStoreKeys, newKey) {
-			result.Added[newKey] = app.keys[newKey]
-		}
-	}
-	return result, nil
-}
 
-// migrateModules performs migrations on existing modules that have registered migrations
-// between versions and initializes the state of new modules for the specified app version.
-func (app *App) migrateModules(ctx sdk.Context, fromVersion, toVersion uint64) error {
-	return app.manager.RunMigrations(ctx, app.configurator, fromVersion, toVersion)
+	// TODO: check if needed, it is no more there
+	// res.Timeouts.TimeoutCommit = app.getTimeoutCommit(currentVersion)
+	// res.Timeouts.TimeoutPropose = appconsts.GetTimeoutPropose(currentVersion)
+	return res, nil
```

Comment:
is there an alternative mechanism to change the timeouts in a rolling fashion?

Reply:
Need to dig a bit more into this one, will try and investigate this today!

Reply:
I don't have full context on why these were removed. Were the additions of these to `State` in core done specifically by the celestia-core fork? 

As I understand these are config.toml configurations in tm, and could be set by a node op so you may end up with differing configurations node by node (?) 

Is the intention for timeout commit to be controlled by the state machine so all validators are aligned?